agent/pom.xml

```diff
@@ -14,4 +14,57 @@
     <version>1.0-SNAPSHOT</version>
     <packaging>jar</packaging>
     <name>Remoting Kafka Plugin Agent</name>
+    <build>
+        <plugins>
+            <plugin>
+                <artifactId>maven-assembly-plugin</artifactId>
+                <executions>
+                    <execution>
+                        <phase>package</phase>
+                        <goals>
+                            <goal>single</goal>
+                        </goals>
+                    </execution>
+                </executions>
+                <configuration>
+                    <archive>
+                        <manifest>
+                            <addClasspath>true</addClasspath>
+                            <mainClass>io.jenkins.plugins.remotingkafka.Agent</mainClass>
+                        </manifest>
+                    </archive>
+                    <descriptorRefs>
+                        <descriptorRef>jar-with-dependencies</descriptorRef>
+                    </descriptorRefs>
+                </configuration>
+            </plugin>
+        </plugins>
+    </build>
+    <dependencies>
+        <dependency>
+            <groupId>org.kohsuke.args4j</groupId>
+            <artifactId>args4j-maven-plugin</artifactId>
```

Comment:
Likely you wanted to include:

```
<groupId>args4j</groupId>
<artifactId>args4j</artifactId>
```

Here you include a Maven plugin, which likely pulls in many undesired dependencies